src/main/java/com/google/impactdashboard/data/recommendation/RecommendationAction.java

```diff
@@ -0,0 +1,23 @@
+package com.google.impactdashboard.data.recommendation;
+
+import com.google.auto.value.AutoValue;
+
+/** Represents an action taken by a GCP Recommendation. */
+@AutoValue
+public abstract class RecommendationAction {
+
+  public abstract String getAffectedAccount();
+  public abstract String getPreviousRole();
+  public abstract String getNewRole();
+
+  /** 
+   * Creates a {@code RecommendationAction} representing an action taken by a 
+   * GCP Recommendation that changed the role of user {@code affectedAccount} 
+   * from {@code previousRole} to {@code newRole}. If the user's role was removed 
+   * instead of replaced, then {@code newRole} should be the empty string. 
```

Comment:
I've found that over time, it's important to directly identify which actions are of what type. This is particularly true of data migrations. 

Instead of adding in data patterns, that eng have to re-interpret at every layer of the stack, it's better to qualify this on the data objects themselves. 

Please add an enum describing the type of action.

Reply:
Sure thing!